packages/core/admin/admin/src/content-manager/components/BlocksEditor/BlocksInput/index.js

```diff
@@ -48,12 +50,36 @@ const BlocksInput = ({ readOnly }) => {
   const blocks = useBlocksStore();
   const renderElement = React.useCallback((props) => baseRenderElement(props, blocks), [blocks]);
 
+  const handleEnter = () => {
+    const selectedNode = editor.children[editor.selection.anchor.path[0]];
+    const selectedBlock = Object.values(blocks).find((block) => block.matchNode(selectedNode));
+
+    // Check if there's an enter handler for the selected block
+    if (selectedBlock.handleEnterKey) {
+      selectedBlock.handleEnterKey(editor);
+    } else {
+      // If not, insert a new paragraph
+      Transforms.insertNodes(editor, {
```

Comment:
Just asking, if you are already in a paragraph it creates a new item in the children array of the paragraph correct? not a brand new paragraph node

Reply:
no it should create a brand new node

Reply:
I don't think so, watch this video 

https://github.com/strapi/strapi/assets/2589748/f5b8c86b-4d48-4083-ab69-3853cdd0d031

Reply:
that's an issue indeed! I'll get on it now